src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/CertificateRequest.cs

```diff
@@ -365,6 +365,85 @@ public X509Certificate2 CreateSelfSigned(DateTimeOffset notBefore, DateTimeOffse
             if (!issuerCertificate.HasPrivateKey)
                 throw new ArgumentException(SR.Cryptography_CertReq_IssuerRequiresPrivateKey, nameof(issuerCertificate));
 
+            DateTime notBeforeLocal = notBefore.LocalDateTime;
+            if (notBeforeLocal < issuerCertificate.NotBefore)
+            {
+                throw new ArgumentException(
+                    SR.Format(
+                        SR.Cryptography_CertReq_NotBeforeNotNested,
+                        notBeforeLocal,
+                        issuerCertificate.NotBefore),
+                    nameof(notBefore));
+            }
+
+            DateTime notAfterLocal = notAfter.LocalDateTime;
+
+            // Round down to the second, since that's the cert accuracy.
```

Comment:
Certificates have one-second accuracy (it's stored as yyMMddHHmmss'Z', or yyyyMMddHHmmss'Z').

As the comment in the code suggests, the problem is the following code:

```C#
DateTimeOffset now = DateTimeOffset.UtcNow;
DateTimeOffset notAfter = now.AddDays(90);

using (var root = MakeRootCert(now, notAfter))
{
    ...
    using (var issued = certReq.Create(root, now, notAfter, serial))
    {
        ...
    }
}
```

The notAfter almost certainly contains some sub-second value, and the first one got rounded down (by the nature of certs), so now it's comparing 2017-04-24T15:35:21.333 to 2017-04-24T15:35:21.  Once it goes into a cert they're the same thing, so it shouldn't fail.